dspace-api/src/main/resources/spring/spring-dspace-addon-import-services.xml

```diff
@@ -63,6 +75,42 @@
           </property>
     </bean>
 
+    <bean id="CsvImportService"
+          class="org.dspace.importer.external.csv.service.CharacterSeparatedImportMetadataSourceServiceImpl" scope="singleton">
+          <property name="metadataFieldMap" ref="csvMetadataFieldMap"></property>
+          <property name="skipLines" value="1" />
+          <property name="supportedExtensions">
+               <list>
+                     <value>csv</value>
+               </list>
+          </property>
+    </bean>
+
+    <bean id="TsvImportService"
+          class="org.dspace.importer.external.csv.service.CharacterSeparatedImportMetadataSourceServiceImpl" scope="singleton">
+          <property name="importSource" value="TsvMetadataSource" />
+          <property name="separator" value="9" />
```

Comment:
Why use a number of separator, can we not enter a \t here ? 9 is a bit unreadable.

Reply:
Personally don't find / know any way to inject char in Spring's beans using the xml config. That's because Spring use them as String by default and there isn't any valid wrapper for char, nor a way to create it. A solution could be create a map of allowed strings and translate the one in input into its ascii value. Something like ["!" -> 1], ["#" -> 3] and so on

Reply:
There does seem to be a `String` -> `Character` converter in the default conversion factory:  https://stackoverflow.com/a/42041652/2916377

Reply:
It doesn't do the trick because just operate on a single character, doesn't work with, for example, tab writes as "\t".

https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/main/java/org/springframework/core/convert/support/StringToCharacterConverter.java

Reply:
What a surprising deficiency.  We could build our own converter that understands normal Java string escapes.

Reply:
We could, but I think the only way to make that is to switch over a list of cases. Something like:

```
if (string.lenght() == 1){  
   return normalCharConverter(character);  
} else {  
   return mapOfStringToChar.get(character);  
}
```
where mapOfStringToChar looks like:  
```
[   
   { "\t", 9 },  
   { "\n", 10},  
   ...  
]  
```

Reply:
https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringEscapeUtils.html#unescapeJava(java.lang.String)

Reply:
We checked the previous implementation BTE implementation and we found that it was configured using unicode escape number, i.e \u0009 that is indeed much clear. We have applied the same solution.